library/common/engine.cc

```diff
@@ -154,6 +154,19 @@ envoy_status_t Engine::recordGaugeSub(const std::string& elements, uint64_t amou
   return ENVOY_FAILURE;
 }
 
+envoy_status_t Engine::recordHistogramDurationMs(const std::string& elements, uint64_t amount) {
+  if (server_ && client_scope_) {
+    std::string name = Stats::Utility::sanitizeStatsName(elements);
+    server_->dispatcher().post([this, name, amount]() -> void {
+      Stats::Utility::histogramFromElements(*client_scope_, {Stats::DynamicName(name)},
```

Comment:
Why don't we use the *FromStatNames methods in utility.h? I see that we are passing a dynamic name, but I'm not sure why it needs to be a dynamic name.

Reply:
if i remember it correctly, it's bc the other way(w/o dynamic name) it requires changes on the c++ layer, which we might/could port to the platform layer, it adds complexity tho, there was a discussion on this when counter was introduced

Reply:
Ah got it. Thanks for the context - I'll stick to dynamic names then.